src/Iterators/LDAPIterator.php

```diff
@@ -0,0 +1,294 @@
+            if ($key === 'count') {
+                continue;
+            }
+
+            $result[$key] = $this->getConvertedEntry($entry);
+        }
+
+        return $result;
+    }
+    private function getConvertedEntry(array $entry)
+    {
+        $result = [];
+
+        foreach ($entry as $key => $value) {
+            if (is_int($key)) {
+                continue;
+            }
+            if ($key === 'count') {
+                continue;
+            }
+
+            if (isset($value['count'])) {
+                unset($value['count']);
+            }
+
+            $result[$key] = $value;
+        }
+
+        if ($this->getResolveRangedAttributes() === true) {
+            $result = $this->resolveRangedAttributes($result);
+        }
+
+        return $result;
+    }
+    private function resolveRangedAttributes(array $row)
+    {
+        $result = [];
+        foreach ($row as $key => $value) {
+            $keyExploded = explode(';range=', $key);
+
+            if (count($keyExploded) === 2) {
+                $range = explode('-', $keyExploded[1]);
+                $offsetAndLimit = (int) $range[1] + 1;
+
+                $result[$keyExploded[0]] = array_merge($value, $this->getAttributeRecursive($row['dn'], $keyExploded[0], $offsetAndLimit, $offsetAndLimit));
+            } else {
+                $result[$key] = $value;
+            }
+        }
+
+        return $result;
+    }
+    private function getAttributeRecursive(string $dn, string $attrName, int $offset, int $maxPerRequest)
+    {
+        $attributeValue = [];
+
+        $limit = $offset + $maxPerRequest - 1;
+        $searchedAttribute = $attrName . ';range=' . $offset . '-' . $limit;
+
+        $ldap = $this->getLdap();
+        $entry = $ldap->getEntry($dn, [
+            $searchedAttribute
+        ], true);
+        foreach ($entry as $key => $value) {
+            // skip DN and other fields (if returned)
+            if (stripos($key, $attrName) === false) {
+                continue;
+            }
+
+            $attributeValue = $value;
+
+            // range result (pagination)
+            $keyExploded = explode(';range=', $key);
+
+            $range = explode('-', $keyExploded[1]);
+            $rangeEnd = (int) $range[1];
+
+            if ($range[0] == $offset && $range[1] == $limit) {
+                // more pages, there are more pages to fetch
+                $attributeValue = array_merge($attributeValue, $this->getAttributeRecursive($dn, $attrName, $rangeEnd + 1, $maxPerRequest));
+            }
+        }
+
+        return $attributeValue;
+    }
+    public function current()
+    {
+        if (! is_array($this->current)) {
+            $this->rewind();
+        }
+        if (! is_array($this->current)) {
+            return;
+        }
+
+        return $this->current;
+    }
+    public function key()
+    {
+        if (! is_array($this->current)) {
+            $this->rewind();
+        }
+        if (! is_array($this->current)) {
```

Comment:
There's quite a few style issues in this file I think, e.g. no spacing between methods. A phpcbf run with PSR-2 should fix them all though

Reply:
Weird - I ran phpcs over this with the phpcs.xml.dist from framework and it didn't complain about this line. I've fixed the obvious ones at least.